core/src/Lender.sol

```diff
@@ -208,12 +208,14 @@ contract Lender is Ledger {
         // Accrue interest and update reserves
         (Cache memory cache, ) = _load();
 
-        // Convert `amount` to `units`
-        units = amount.mulDivUp(BORROWS_SCALER, cache.borrowIndex);
+        unchecked {
+            // Convert `amount` to `units`
+            units = (amount * BORROWS_SCALER) / cache.borrowIndex;
```

Comment:
Two reasons why it's okay to _not_ round up:
- As shown in the new `test_math_borrowUnitsUniqueness` test, `BORROWS_SCALER` is large enough that any change in `amount` will cause a corresponding change in `units`, and `units` is never 0 unless `amount` also is. In other words, we're not losing precision
- There's no free money glitch, because the `units` in the `repay` function are computed in the exact same way. So if you borrow some `amount`, the only way to repay all `units` of debt is to repay that same `amount`.

Ok, but why did I bother changing it at all?

Well one of the key attributes of this PR is that `lender.borrowBalance` is more precise. If you call `repay(borrowBalance(someAddr), someAddr)`, it is now guaranteed that the debt will be zeroed out -- previously it was just kinda close-ish. I had to choose floor/ceil very carefully to get this to work.